File: sockets/socket_server.py
# simple_ws_server.py
import asyncio
import websockets

async def handler(websocket):
    async for message in websocket:
        if message == "ping":
            await websocket.send("pong")

async def main():
    async with websockets.serve(handler, "0.0.0.0", 5001):
        await asyncio.Future()  # run forever

if __name__ == "__main__":
    asyncio.run(main())

File: sockets/socket_server.js
const WebSocket = require('ws');

const server = new WebSocket.Server({ port: 3001 });

server.on('connection', (socket) => {
  console.log('Client connected');
  
  socket.on('message', (message) => {
    if (message.toString() === 'ping') {
      socket.send('pong');
    }
  });

  
});

console.log('WebSocket server running on ws://localhost:3001');